### ISSUE TITLE
Memory leak?

### ISSUE THREAD

**Issue by jdmarsh** [devnixs/ODataAngularResources#99]
I noticed a memory leak within our application and tracked it to inside or around my usage of this component.
I have some jsfiddles set up to demonstrate.
If you take a look here: [http://jsfiddle.net/h22f7596/164/show/](http://jsfiddle.net/h22f7596/164/show/) Doing a single odata query lands us at a stable memory usage. (around 63k on my machine)
I then have a fiddle set up where it performs a request to the same end point every 0.5 seconds for 60 seconds. The fiddle is here: [http://jsfiddle.net/h22f7596/163/show/](http://jsfiddle.net/h22f7596/163/show/)
In this example, memory usage keeps climbing as we request more and more data (as expected) however when the 60 seconds are up, the memory usage never decreases back to the stable value seen from the first example above. (It settled to around 84k on my machine)
I performed one last test at: [http://jsfiddle.net/h22f7596/165/show/](http://jsfiddle.net/h22f7596/165/show/) This one doesn't do anything with the received data (nothing on the scope). Here the memory doesn't get released again.

I may be missing something, what do you think?

**Comment by kyse** [comment]
Hmm, initial thought would be turn off persistence and test again.  Perhaps I did some bad stuff with it.  But your tests are just creating a new odataresource object, not utilizing any of the refresh functionality.  Also maybe test if you get the same results without using the odata version, ie odata().query() route, ie determine if the recursive odata querying object constructor has anything to do with it?

**Comment by jdmarsh** [comment]
My personal project experiences memory problems without creating a new odata resource. I'll do more testing tomorrow and ill let you know how it goes.

**Comment by jdmarsh** [comment]
creating the odata resource only once doesn't seem to help, The test at [http://jsfiddle.net/h22f7596/169/show/](http://jsfiddle.net/h22f7596/169/show/) still has the issue.

I took out the filters and just did odata().query() and found that the memory increase is much much slower. The fiddle is here: [http://jsfiddle.net/h22f7596/170/show/](http://jsfiddle.net/h22f7596/170/show/)

**Comment by jdmarsh** [comment]
Looks like the memory increases depending on the size of the data set. Specifically the resourceStore array gets larger and larger the more requests are performed. What is the purpose of the resourceStore and is there a way to empty it?

**Comment by kyse** [comment]
I added in a resourceStore to capture existing settings for the odataAngularResource, so allow providing a way to expose in the promise chain a way inject corrective measures for errors.  Take a look at the tests for examples.

And yes, I think that would definitely be an issue with garbage collection, I need to add in some logic to handle removing/clearing it when the resource is no longer needed.

**Comment by kyse** [comment]
Just want to write a quick update.  I been very busy with work, but I have not forgotten about needing to look into this.

**Comment by jdmarsh** [comment]
@kyse Do you know how you would go about fixing this? I can take a look at it if you could point me in the right direction.
